src/Sarif.Converters/CisCatConverter.cs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using Microsoft.CodeAnalysis.Sarif.Converters.CisCatObjectModel;
+
+namespace Microsoft.CodeAnalysis.Sarif.Converters
+{
+    public class CisCatConverter : ToolFileConverterBase
+    {
+        private readonly LogReader<CisCatReport> logReader;
+
+        public CisCatConverter()
+        {
+            logReader = new CisCatReportReader();
+        }
+
+        public override string ToolName => ToolFormat.CisCat;
+
+        public override void Convert(Stream input, IResultLogWriter output, OptionallyEmittedData dataToInsert)
+        {
+            input = input ?? throw new ArgumentNullException(nameof(input));
+            output = output ?? throw new ArgumentNullException(nameof(output));
+
+            //Read CIS CAT data
+            CisCatReport log = logReader.ReadLog(input);
+
+            //Top level run object for the scan data
+            var run = new Run();
+
+            //Set the tool details
+            run.Tool = new Tool();
+            run.Tool.Driver = CreateDriver(log);
+
+            //Set the list of tool rules
+            run.Tool.Driver.Rules = new List<ReportingDescriptor>();
+            foreach (CisCatRule rule in log.Rules)
+            {
+                run.Tool.Driver.Rules.Add(CreateReportDescriptor(rule));
```

Comment:
>run.Tool.Driver.Rules.Add(CreateReportDescriptor(rule)); [](http://example.com/codeflow?start=16&length=56)

Suggest to only log a rule if its referenced by a result and avoid logging duplicated rules.

This can be done in the results loop below and checking if the rule already exists in the `log.Rules' list.